source/renderer/app/components/wallet/wallet-import/WalletSelectImportDialog.js

```diff
@@ -145,15 +152,45 @@ export default class WalletSelectImportDialog extends Component<Props> {
     return walletStatus;
   };
 
-  getWalletStatusIcon = (wallet: ExportedByronWallet) => {
-    const { nameValidator, onToggleWalletImportSelection } = this.props;
+  getWalletStatusIcon = (wallet: ExportedByronWallet, index: number) => {
+    const {
+      nameValidator,
+      onToggleWalletImportSelection,
+      isMaxNumberOfWalletsReached,
+    } = this.props;
     let statusIcon;
     if (
       wallet.import.status === WalletImportStatuses.UNSTARTED ||
       wallet.import.status === WalletImportStatuses.PENDING ||
       wallet.import.status === WalletImportStatuses.ERRORED
     ) {
-      const disabled = wallet.name === null || !nameValidator(wallet.name);
+      const invalidWalletName =
+        wallet.name === null || !nameValidator(wallet.name);
+      const walletNotSelectable =
+        isMaxNumberOfWalletsReached &&
```

Comment:
@tomothespian here we should check the number of selected wallets and not only if the max number has been reached...